EventFilter/HGCalRawToDigi/src/HGCalUnpacker.cc

```diff
@@ -84,17 +104,144 @@ void HGCalUnpacker::parseFEDData(unsigned fedId,
+            digis.view()[denseIdx].cm() = cmSum;
+            digis.view()[denseIdx].flags() = 0;
+            iBit += erxBodyBits_[code];
+          }
+          iword += iBit / 32;
+          if (iBit % 32 != 0) {
+            iBit += 1;
+          }
+        }
+      } else {
+        // Passthrough ECON-D
+        std::cout << "Passthrough ECON-D, erxMax=" << erxMax << "enabledErx= " << enabledErx << std::endl;
+        for (uint32_t erxIdx = 0; erxIdx < erxMax; erxIdx++) {
+          // check if the eRx is enabled
+          if ((enabledErx >> erxIdx & 1) == 0) {
+            continue;
+          }
+          std::cout << "fedId = " << fedId << ", captureblockIdx = " << captureblockIdx << ", econdIdx = " << econdIdx
+                    << ", erxIdx=" << erxIdx << std::endl;
+
+          // check if the eRx sub-packet is empty (the "F" flag in the eRx sub-packet header)
+          if (((econd_payload[iword] >> ECOND_FRAME::ERXFORMAT_POS) & ECOND_FRAME::ERXFORMAT_MASK) == 1) {
+            std::cout << "eRx " << erxIdx << " is empty" << std::endl;
+            iword += 1;  // length of an empty eRx header (32 bits)
+            continue;    // go to the next eRx
+          }
+
+          // erx header
+          uint16_t cmSum = ((econd_payload[iword] >> ECOND_FRAME::COMMONMODE0_POS) & ECOND_FRAME::COMMONMODE0_MASK) +
+                           ((econd_payload[iword + 1] >> ECOND_FRAME::COMMONMODE1_POS) & ECOND_FRAME::COMMONMODE1_MASK);
+          uint64_t erxHeader = ((uint64_t)econd_payload[iword] << 32) | ((uint64_t)econd_payload[iword + 1]);
+          uint32_t denseIdx = moduleIndexer.getIndexForModuleErx(fedId, globalECONDIdx, erxIdx);
+          std::cout << ", erx_headers = 0x" << std::hex << std::setfill('0') << std::setw(16) << erxHeader
+                    << ", cmSum = " << std::dec << cmSum << std::endl;
+          iword += 2;
+
+          // parse erx body (channel data)
+          for (uint32_t channelIdx = 0; channelIdx < maxChPerErxNonCM; channelIdx++) {
+            // always increment the denseIdx
+            ++denseIdx;
+
+            // check if the channel has data
+            if (((erxHeader >> channelIdx) & 1) == 0) {
+              continue;
+            }
+
+            // TODO: Check what to do with pass through ECOND
+            // can re-use the implementation in HGCROCChannelDataFrame
+            // but needs to know whether the ROCs are in characterization mode or not
+            digis.view()[denseIdx].tctp() = (econd_payload[iword] >> 30) & 0b11;
+            digis.view()[denseIdx].adcm1() = 0;
+            digis.view()[denseIdx].adc() = (econd_payload[iword] >> 20) & 0b1111111111;
+            digis.view()[denseIdx].tot() = (econd_payload[iword] >> 10) & 0b1111111111;
+            digis.view()[denseIdx].toa() = econd_payload[iword] & 0b1111111111;
+            digis.view()[denseIdx].cm() = cmSum;
+            digis.view()[denseIdx].flags() = 0;
```

Comment:
@pfs Any suggestions on this?

Reply:
There'll be a parameter from conditions to decide what to do right?